### PR TITLE
bumps auxlua to 1.4.0

### PR DESCRIPTION
**Pull request by Y0SH1M4S73R**
## About The Pull Request

Mothblocks recently talked about wanting an auxlua function to check if the lua state was about to overrun its execution limit, and that was something I had been thinking about for a long time prior, but never bothered to actually implement until it was brought up. So I did just that. The documentation has also been updated to include a description of how the execution limit works and how to use the new function, `over_exec_time`

## Why It's Good For The Game

Allows for lua loops of indeterminate length to do as much work as they reasonably can in a single tick without setting off the execution limiter and erroring out.

## Changelog

:cl:
admin: Adds a new function for admin lua scripting, "over_exec_time", for checking if lua code is running close to the execution limit. Details are available in the lua editor's help menu.
admin: The execution limit is described in detail in the lua editor's help menu.
/:cl:

### CODE REVIEW

File: code/modules/admin/verbs/lua/README.md [hunk 1]
```diff
@@ -105,6 +105,18 @@ A weak reference to DM's `usr`. As a rule of thumb, this is a reference to the m
 
 ---
 
+## Execution Limit
+
+In order to prevent freezing the server with infinite loops, auxlua enforces an execution limit, defaulting to 100ms. When a single lua state has been executing for longer than this limit, it will eventually stop and produce an error.
+
+To avoid exceeding the execution limit, call `sleep()` or `coroutine.yield()` before the execution limit is reached.
+
+### over_exec_usage(fraction)
```
Comment (Mothblocks) [inline code/modules/admin/verbs/lua/README.md:114]:
```suggestion
### over_exec_usage(fraction = 0.95)
```

**Review by Mothblocks — APPROVED** [review]

**Comment by Mothblocks** [comment]
Hey wait, looks like all the checks are failing for auxlua when they don't on other PRs.

image

**Comment by Y0SH1M4S73R** [comment]
Okay I don't know why this is going wrong, but I'll have to make a new version that fixes all that. I'll take care of it tomorrow.

**Comment by Y0SH1M4S73R** [comment]
Ok so the auxlua CI failure was due to the fact that the linux build of auxlua was done on ubuntu-latest instead of ubuntu-20.04. I had to modify the auxlua build workflow to account for this, and will be pushing an empty commit to re-run CI once the build finishes.

**Pull request merged**